src/pmi/src/pmi_wire.c

```diff
@@ -14,7 +14,7 @@
 #define IS_SPACE(c) ((c) == ' ')
 #define IS_EOS(c) ((c) == '\0')
 #define IS_EOL(c) ((c) == '\n' || (c) == '\0')
-#define IS_KEY(c) (isalnum(c) || (c) == '_' || (c) == '-' || (c) == '/')
+#define IS_KEY(c) (!(IS_SPACE(c) || IS_EOL(c) || (c) == '='))
```

Comment:
Will users get an error if a put or get is attempted with a bad character?

Reply:
It likely will get a parse error, which will become a PMI error and then an MPI error.